dist/tools/openocd/openocd.sh

```diff
@@ -178,6 +178,13 @@ do_flash() {
 do_debug() {
     test_config
     test_elffile
+
+    # Configure halt command: prepend a reset if required by the board config
+    local _pre_reset_halt=''
```

Comment:
@kYc0o, that's why I changed my code that now update a local variable by checking if the global variable is set.

Reply:
By the way, this strategy was inspired by the use of PRE_FLASH_CHECK_SCRIPT. The idea is exactly the same.

Reply:
> By the way, this strategy was inspired by the use of PRE_FLASH_CHECK_SCRIPT. The idea is exactly the same.

That's a condition to execute a whole script and analysing its output, not just setting a variable. What I'm trying to avoid here is the declaration of variables and conditions which are IMHO not necessary, and by simply declaring the variable in makefiles/vars.inc.mk and setting it in your board Makefile it's enough. It's even more flexible since for other boards (which we might not be aware of yet) other commands might be necessary, thus you can just set this variable for your needs e.g. `PRE_DEBUG_RESET_HALT += reset halt reset` if you need to reset twice or whatever.

Reply:
Looks like we disagree